From 3ab2316b733183890b52fcec4f34aee87376d297 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
Date: Mon, 23 May 2022 11:11:01 +0300
Subject: [PATCH 2/2] debugcc: add block names to all platforms

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
---
 sc8280xp.c | 2 ++
 sdm845.c   | 5 +++++
 sm8350.c   | 4 ++++
 sm8450.c   | 4 ++++
 4 files changed, 15 insertions(+)

diff --git a/sc8280xp.c b/sc8280xp.c
index 2f0ef2a2cb69..ad10a9db96d3 100644
--- a/sc8280xp.c
+++ b/sc8280xp.c
@@ -61,6 +61,7 @@ static struct debug_mux gcc = {
 static struct debug_mux disp0_cc = {
 	.phys = 0xaf00000,
 	.size = 0x20000,
+	.block_name = "disp0",
 
 	.enable_reg = 0x500c,
 	.enable_mask = BIT(0),
@@ -76,6 +77,7 @@ static struct debug_mux disp0_cc = {
 static struct debug_mux disp1_cc = {
 	.phys = 0x22100000,
 	.size = 0x20000,
+	.block_name = "disp1",
 
 	.enable_reg = 0x500c,
 	.enable_mask = BIT(0),
diff --git a/sdm845.c b/sdm845.c
index 8abade19cfdd..a66abf90346d 100644
--- a/sdm845.c
+++ b/sdm845.c
@@ -71,6 +71,7 @@ static struct debug_mux gcc = {
 static struct debug_mux cam_cc = {
 	.phys = 0xad00000,
 	.size = 0x10000,
+	.block_name = "cam",
 
 	.enable_reg = 0xc008,
 	.enable_mask = BIT(0),
@@ -85,6 +86,7 @@ static struct debug_mux cam_cc = {
 static struct debug_mux cpu = {
 	.phys = 0x17970000,
 	.size = 4096,
+	.block_name = "cpu",
 
 	.enable_reg = 0x18,
 	.enable_mask = BIT(0),
@@ -101,6 +103,7 @@ static struct debug_mux cpu = {
 static struct debug_mux disp_cc = {
 	.phys = 0xaf00000,
 	.size = 0x10000,
+	.block_name = "disp",
 
 	.enable_reg = 0x600c,
 	.enable_mask = BIT(0),
@@ -115,6 +118,7 @@ static struct debug_mux disp_cc = {
 static struct debug_mux gpu_cc = {
 	.phys = 0x5090000,
 	.size = 0x9000,
+	.block_name = "gpu",
 
 	.enable_reg = 0x1100,
 	.enable_mask = BIT(0),
@@ -129,6 +133,7 @@ static struct debug_mux gpu_cc = {
 static struct debug_mux video_cc = {
 	.phys = 0xab00000,
 	.size = 0x10000,
+	.block_name = "video",
 
 	.enable_reg = 0xa58,
 	.enable_mask = BIT(0),
diff --git a/sm8350.c b/sm8350.c
index 58a41654c0b2..ecf8ea94afeb 100644
--- a/sm8350.c
+++ b/sm8350.c
@@ -71,6 +71,7 @@ static struct debug_mux gcc = {
 static struct debug_mux cam_cc = {
 	.phys = 0xad00000,
 	.size = 0x10000,
+	.block_name = "cam",
 
 	.enable_reg = 0xd008,
 	.enable_mask = BIT(0),
@@ -86,6 +87,7 @@ static struct debug_mux cam_cc = {
 static struct debug_mux disp_cc = {
 	.phys = 0xaf00000,
 	.size = 0x10000,
+	.block_name = "disp",
 
 	.enable_reg = 0x500c,
 	.enable_mask = BIT(0),
@@ -101,6 +103,7 @@ static struct debug_mux disp_cc = {
 static struct debug_mux gpu_cc = {
 	.phys = 0x3d90000,
 	.size = 0x9000,
+	.block_name = "gpu",
 
 	.enable_reg = 0x1100,
 	.enable_mask = BIT(0),
@@ -116,6 +119,7 @@ static struct debug_mux gpu_cc = {
 static struct debug_mux video_cc = {
 	.phys = 0xaaf0000,
 	.size = 0x10000,
+	.block_name = "video",
 
 	.enable_reg = 0xebc,
 	.enable_mask = BIT(0),
diff --git a/sm8450.c b/sm8450.c
index ba1c9857ec24..a04de128ae4d 100644
--- a/sm8450.c
+++ b/sm8450.c
@@ -82,6 +82,7 @@ static struct debug_mux apss_cc = {
 static struct debug_mux cam_cc = {
 	.phys = 0xade0000,
 	.size = 0x20000,
+	.block_name = "cam",
 
 	.enable_reg = 0x14008,
 	.enable_mask = BIT(0),
@@ -97,6 +98,7 @@ static struct debug_mux cam_cc = {
 static struct debug_mux disp_cc = {
 	.phys = 0xaf00000,
 	.size = 0x20000,
+	.block_name = "disp",
 
 	.enable_reg = 0xd00c,
 	.enable_mask = BIT(0),
@@ -112,6 +114,7 @@ static struct debug_mux disp_cc = {
 static struct debug_mux gpu_cc = {
 	.phys = 0x3d90000,
 	.size = 0xa000,
+	.block_name = "gpu",
 
 	.enable_reg = 0x9274,
 	.enable_mask = BIT(0),
@@ -127,6 +130,7 @@ static struct debug_mux gpu_cc = {
 static struct debug_mux video_cc = {
 	.phys = 0xaaf0000,
 	.size = 0x10000,
+	.block_name = "video",
 
 	.enable_reg = 0x80ec,
 	.enable_mask = BIT(0),
-- 
2.35.1

