From 658d2022e3a31c916deec1aa9e35a4affc5e3d05 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
Date: Fri, 23 Sep 2022 18:01:00 +0300
Subject: [PATCH] msm8996: add block names

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
---
 msm8996.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/msm8996.c b/msm8996.c
index bde2a95707ef..aa86f8351351 100644
--- a/msm8996.c
+++ b/msm8996.c
@@ -72,6 +72,7 @@ static struct debug_mux gcc = {
 static struct debug_mux mmss_cc = {
 	.phys = 0x8c0000,
 	.size = 0xb00c,
+	.block_name = "mmss",
 
 	.enable_reg = 0x900,
 	.enable_mask = BIT(16),
@@ -112,6 +113,7 @@ static void cpu_postmeasure(struct debug_mux *mux)
 static struct debug_mux cpu_cc = {
 	.phys = 0x09820000,
 	.size = 0x1000,
+	.block_name = "cpu",
 
 	.mux_reg = 0x78,
 	.mux_mask = 0xff << 8,
-- 
2.35.1

